services/audit-service/src/main/java/datawave/microservice/audit/replay/ReplayController.java

```diff
@@ -0,0 +1,726 @@
+                    @ApiParam("The file URI to use, relative to the path") @RequestParam(defaultValue = "") String fileUri,
+                    @ApiParam(value = "The number of messages to send per second", defaultValue = "100") @RequestParam(defaultValue = "100") Long sendRate,
+                    @ApiParam(value = "Indicates whether files from an unfinished audit replay should be included",
+                                    defaultValue = "false") @RequestParam(defaultValue = "false") boolean replayUnfinishedFiles,
+                    HttpServletResponse response) {
+        
+        log.info("Creating and starting audit replay with params: path=" + path + ", fileUri=" + fileUri + ", sendRate=" + sendRate + ", replayUnfinishedFiles="
+                        + replayUnfinishedFiles);
+        
+        String resp;
+        
+        // only create if the send rate is valid
+        if (sendRate >= 0) {
+            String id = UUID.randomUUID().toString();
+            
+            Status status = statusCache.create(id, path, (!fileUri.isEmpty()) ? fileUri : FileSystem.getDefaultUri(config).toString(), sendRate,
+                            replayUnfinishedFiles);
+            runningReplays.put(id, start(status));
+            
+            log.info("Created and started audit replay [" + status + "]");
+            
+            resp = id;
+        } else {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            resp = "Send rate must be >= 0";
+        }
+        
+        log.info(resp);
+        
+        return resp;
+    }
+    
+    /**
+     * Starts an audit replay
+     *
+     * @param id
+     *            The audit replay id
+     * @return status, indicating whether the audit replay was started successfully
+     */
+    @ApiOperation(value = "Starts an audit replay.")
+    @RolesAllowed({"Administrator", "JBossAdministrator"})
+    @RequestMapping(path = "/{id}/start", method = org.springframework.web.bind.annotation.RequestMethod.POST)
+    public String start(@ApiParam(value = "The audit replay id") @PathVariable String id, HttpServletResponse response) {
+        
+        log.info("Starting audit replay with id " + id);
+        
+        Status status = status(id, replayProperties.isPublishEvents());
+        
+        String resp;
+        // if the state is 'created', we can run the replay
+        if (status != null) {
+            if (status.getState() == CREATED) {
+                RunningReplay runningReplay = start(status);
+                if (runningReplay != null) {
+                    runningReplays.put(id, runningReplay);
+                    resp = "Started audit replay with id " + id;
+                } else {
+                    resp = "Cannot start audit replay with id " + id;
+                    response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
Bad request is documented "The server cannot or will not process the request due to an apparent client error (e.g., malformed request syntax, size too large, invalid request message framing, or deceptive request routing)." Using this as a response code doesn't seem to fit here.

Reply:
The next commit will have this changed to SC_INTERNAL_SERVER_ERROR.

Reply:
I'm sure there's a suitable one in https://en.wikipedia.org/wiki/List_of_HTTP_status_codes